marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/steps/CustomStepControllerTest.java

```diff
@@ -0,0 +1,102 @@
+package com.marklogic.hub.central.controllers.steps;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+import com.marklogic.hub.dataservices.StepService;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.security.access.AccessDeniedException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+public class CustomStepControllerTest extends AbstractStepControllerTest {
+
+    private final static String PATH = "/api/steps/custom";
+    
+    /* used to create custom steps */
+    private final static String DEFAULT_JSON = "{"
+    		+ " \"name\": \"myCustomStep\" , "
+    		+ " \"description\": \"\" , "
+    		+ " \"selectedSource\": \"collection\" , "
+    		+ " \"sourceQuery\": \"cts.collectionQuery('test')\" , "
+    		+ " \"targetEntityType\": \"http://example.org/Customer-0.0.1/Customer\" , "
+    		+ " \"stepDefinitionName\": \"custom\" "
+    		+ "}";
+    private final static ObjectMapper MAPPER = new ObjectMapper();
+
+    public static class CustomStep {
+        public String name;
+        public String description;
+        public String selectedSource;
+        public String sourceQuery;
+        public String targetEntityType;
+        public String stepDefinitionName;
+    }
+
+    public static CustomStep newDefaultCustomStep(String name) {
+        CustomStep step = new CustomStep();
+        step.name = name;
+        step.description = "optional";
+        step.selectedSource = "collection";
+        step.sourceQuery = "cts.collectionQuery('test')";
+        step.targetEntityType = "http://example.org/Customer-0.0.1/Customer";
+        step.stepDefinitionName = "custom";
+        return step;
+    }
+
+    @Test
+    void userCanUpdateStep() throws Exception {
+        installOnlyReferenceModelEntities();
+        loginAsTestUserWithRoles("hub-central-custom-writer");
+
+        JsonNode new_node = MAPPER.readTree(DEFAULT_JSON);
+        StepService.on(getHubClient().getStagingClient()).saveStep("custom", new_node, false, true);
+        
+        /* update the description */
+        putJson(PATH + "/myCustomStep", newDefaultCustomStep("myCustomStep")).andExpect(status().isOk());
+        
+        getJson(PATH).andExpect(status().isOk())
+        .andDo(result -> {
+        	ArrayNode array = (ArrayNode) parseJsonResponse(result).get("stepsWithEntity");
+            assertEquals(2, array.size());

```

Comment:
Why are there 2 steps here ? I only see one step created in the test.

Reply:
I'm not sure.  It's how the response comes back.  Both nodes have entityType and artifacts sub-trees, but the one with an entityType of "Order" doesn't have anything useful.  The next 4 lines is just looking for the subtree that does have the custom step.

Reply:
I should clarify that the Order subtree always seems to be present no matter what type of step it is, nor how many test steps were added -- the number of subtrees in `ArrayNode array` is always 2.
All the actual test steps are in the Customer subtree.  I will add a check to make sure there is only one step there.

Reply:
Please figure it out. I am only seeing one step created here and before every test runs, the db is cleared (except for hub-core-artifacts). So, you should see only one step "myCustomStep" and associated with entity "Customer" that's it.

Reply:
Added clarification